allennlp/semparse/executors/wikitables_variable_free_executor.py

```diff
@@ -0,0 +1,647 @@
+from typing import List, Dict, Tuple, Union
+from collections import defaultdict
+import logging
+
+from overrides import overrides
+
+from allennlp.semparse.worlds.world import ExecutionError
+from allennlp.semparse.contexts.table_question_knowledge_graph import MONTH_NUMBERS
+from allennlp.semparse.executors.executor import Executor, NestedList
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Date:
+    def __init__(self, year: int, month: int, day: int) -> None:
+        self.year = year
+        self.month = month
+        self.day = day
+
+    def __eq__(self, other) -> bool:
```

Comment:
I think it'd be a lot more clear to just call this something else, like `is_compatible()`.  `==` has an expected semantics, and you're changing it.

Reply:
By using the in-built functions, I was able to directly sort the values within argmax and argmin. And what else would you expect from `==` for the `Date` class?

Reply:
I'm just saying that `==` has an expected semantics in python, and it's a bit jarring to me to see it changed like this, so that, e.g., it's no longer transitive.  That's probably fine if this class is only used in this executor, just be sure you document it clearly.

Reply:
Done.